### PR TITLE
update allure report documentation and add cucumber example

### PR DESCRIPTION
**Pull request by anupamck**
## Proposed changes

Minor changes to documentation to update it, given some information pertains to older wdio major versions. 

Note: This is my first ever PR to an open-source project, and is a small start :)

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-allure-reporter/README.md [hunk 1]
```diff
@@ -162,8 +172,8 @@ Screenshots can be attached to the report by using the `takeScreenshot` function
 First set `disableWebdriverScreenshotsReporting: false` in reporter options, then add in afterStep hook:
 
 ```js title="wdio.conf.js"
-afterStep: function (test, scenario, { error, duration, passed }) {
-  if (error) {
+afterStep: function (step, scenario, result, context) {
+  if (result.error) {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/README.md:176]: Any particular reason for this change? It is the same as the previous version.

File: packages/wdio-allure-reporter/README.md [hunk 2]
```diff
@@ -98,6 +98,16 @@ describe('Suite', () => {
 })
 ```
 
+Cucumber example
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/README.md:101]:
```suggestion
Cucumber example:
```

**Review by christian-bromann — COMMENTED** [review]
Hey @anupamck , thanks for taking a stab at this and improving the docs. I have one minor formatting request (which I will auto-merge) and a question.

**Comment by christian-bromann** [comment]
@anupamck please make sure to sign the CLA with the same email address you used for creating the commit. Let me know if you have any troubles with it.

File: packages/wdio-allure-reporter/README.md (hunk 1 above)
Comment (anupamck) [reply packages/wdio-allure-reporter/README.md:176]: Well, these are the parameters included as comments in the `wdio.conf.js` file, in the afterStep hook:
```
     * @param {ITestCaseHookParameter} world            world object containing information on pickle and test step
     * @param {Object}                 result           results object containing scenario results
     * @param {boolean}                result.passed    true if scenario has passed
     * @param {string}                 result.error     error stack if scenario failed
     * @param {number}                 result.duration  duration of scenario in milliseconds
     * @param {Object}                 context          Cucumber World object
```

And they differ from the parameters shown in the current documentation for the allure reporter, where we have the following: `afterStep: function (test, scenario, { error, duration, passed })`

Perhaps this is my ignorance, but I find this difference confusing.

**Comment by anupamck** [comment]
> @anupamck please make sure to sign the CLA with the same email address you used for creating the commit. Let me know if you have any troubles with it.

I'm having troubles here. I didn't include an email address in the commit. I used the same email address linked to this github id. I have already signed it thrice :)

**Comment by anupamck** [comment]
> Not sure how you did the commit, through GitHub or locally. For the latter please have a read [here]

I have included my email address now. Hadn't done this earlier, and it was a local commit. Should I send a new commit?

**Comment by christian-bromann** [comment]
> Should I send a new commit?

Yea, either rebase and create a new commit or close this one and create a new PR.

**Comment by christian-bromann** [comment]
> [@anupamck ](/anupamck)
> ` [include email id](/webdriverio/webdriverio/pull/8325/commits/92e47ec7332facf5287772858ae766dda3ed5aed)`

All commits in this PR need to be made with the correct git email, I suggest that you just start from fresh, make sure your git settings are correct, create a new branch and make a new commit with a new PR

**Pull request closed without merging**